torch/_inductor/codegen/triton.py

```diff
@@ -1490,15 +1490,20 @@ def should_use_persistent_reduction(self) -> bool:
         Heuristic to set self.persistent_reduction and add guards
         if needed.
         """
-        if (
-            not (self.inside_reduction and config.triton.persistent_reductions)
-            or self.cooperative_reduction
-        ):
+        if not (self.inside_reduction and config.triton.persistent_reductions):
             return False
         threshold = {
             ReductionHint.INNER: 1024,
         }.get(self.reduction_hint, 64)
 
+        if self.cooperative_reduction:
+            # The RSPLIT of cooperative reductions means each thread block is operating on fewer elements
+            xnumel, _ = self.numels
+            try:
+                threshold *= 32 // V.graph.sizevars.size_hint(xnumel)
```

Comment:
So `32 // V.graph.sizevars.size_hint(xnumel)` is used to estimate RSPLIT?

Is it expected that this is not that consistent to the similar logic in triton_heuristics.cooperative_reduction :
```
split = max(1, min(target // xnumel, TRITON_MAX_RSPLIT))
```

target is 64 there.

Reply:
Yeah, though I'm intentionally under-estimating here to be conservative.